command_line/expeditor.py

```diff
@@ -0,0 +1,213 @@
+from __future__ import absolute_import, division, print_function
+
+from itertools import groupby
+from tabulate import tabulate
+import copy
+from libtbx.phil import parse
+from dials.util import show_mail_on_error
+from dials.util.options import OptionParser, flatten_reflections, flatten_experiments
+from dials.array_family import flex
+from dxtbx.model.experiment_list import ExperimentList
+
+help_message = """
+
+Utility to wrangle experiments from indexing in preparation for refinement -
+will look to see that the sample is centred, that there are observations for
+the full range of every scan and rewrite the experiment lists if not.
+
+  dials.expeditor indexed.expt indexed.refl
+
+"""
+
+
+phil_scope = parse(
+    """
+output {
+  experiments = expedited.expt
+    .type = path
+    .help = "Filename for fixed experiment list"
+
+  reflections = expedited.refl
+    .type = path
+    .help = "Filename for fixed reflection list"
+}
+"""
+)
+
+
+def model_connectivity(experiments):
+    def model_connectivity_impl(experiments, model):
+        text = [""]
+        text.append("%s:" % model.capitalize())
+        models = getattr(experiments, "%ss" % model)()
+        rows = [[""] + [str(j) for j in range(len(models))]]
+        for j, e in enumerate(experiments):
+            row = ["Experiment %d" % j]
+            for m in models:
+                if getattr(e, model) is m:
+                    row.append("x")
+                else:
+                    row.append(".")
+            rows.append(row)
+        text.append(tabulate(rows, tablefmt="plain"))
+        return text
+
+    if len(experiments) == 1:
+        return ""
+
+    text = []
+    text.append("Experiment / Models")
+    text.extend(model_connectivity_impl(experiments, "detector"))
+    text.extend(model_connectivity_impl(experiments, "crystal"))
+    text.extend(model_connectivity_impl(experiments, "beam"))
+
+    return "\n".join(text)
+
+
+def select_scans_from_reflections(reflections, scan):
+    """Determine a list of valid scans where reflection shoeboxes are seen,
+    from within the bounds of the input scan."""
+
+    bbox = reflections["bbox"].parts()
+
+    z0, z1 = bbox[4], bbox[5]
```

Comment:
```suggestion
    z0, z1 = reflections["bbox"].parts()[4:]
```
might be simpler, since the `bbox` variable above isn't otherwise needed.

Reply:
Or alternatively:
```_, _, _, _, z0, z1 = reflections["bbox"].parts()`